### PR TITLE
17 modernizr update

### PR DESCRIPTION
**Pull request by jhogue**
Relates to Issue #17 

Add our most recent version of the touch-hover mixin to the scaffold. Includes a set of variables for the Modernizr hook. Authors can change the hook in one place if they are using a newer version of modernize. 

Old Modernizr: `no-touch / touch`
New Modernizr: `no-touchevents / touchevents`

### CODE REVIEW

**Comment by bendoh** [comment]
Looks nice, well-commented, etc. /approved

File: base/_typesizes.scss [hunk 1]
```diff
@@ -63,6 +63,18 @@ $type-sizes: (
 // ! ===== Mixins to support pulling out type sizes at breakpoints =====
 
 //
+// Deep getter to go into the $type-sizes array
+// and return with a specific element at a particular breakpoint
+// @jhogue
+//
+// @usage: .bravo { font-size: type-size(h2,large); }
```
Comment (raubin) [inline base/_typesizes.scss]: @jhogue - isn't this backwards to how you've defined this function?

Comment (jhogue) [reply base/_typesizes.scss]: Hang on, thats not a file that I changed in this branch. Let me investigate the source and get back to you.

**Comment by jhogue** [comment]
@kbeck303 @nicholemaureen @johncionci @BrandonDH @syzygy333 

Team… Cionci recently requested a fallback for when Modernizr is not available. I added that as a variable that we can switch true/false depending on the theme, and that lets us continue to write and structure our code in the same way every time. 

I also added documentation for preferred use case, how to use the `disabled` option, and what output to expect. 

Please take a look and lets get this into Master soon so John can use the code for Washfunders.

**Comment by jhogue** [comment]
This will close Issue #17. The variables can be changed depending on the project version of Modernizr.

**Comment by johncionci** [comment]
@jhogue  Looks good from here, nice cleanup and alpha order updates. Did we test? if not, I can pull into WASH tomorrow and try it out.

@kbeck303 @nicholemaureen @BrandonDH @syzygy333
Thoughts?

**Comment by jhogue** [comment]
I tested only by running the code is Sassmeister to work out the kinks and get the sample output. I know it will work and will not produce an error, but I'd like to see you use it in the field to determine if we need to change anything.

File: base/_typesizes.scss [hunk 2]
```diff
@@ -93,6 +106,14 @@ $type-sizes: (
...
 // @jhogue
 //
+// @usage: @media (min-width: bp(medium) ) { @include all-type-sizes(medium); }
+// @returns: @media (min-width: 60em) {
+//	h1 { font-size: 1.6875rem; }
+//	h2 { font-size: 1.5rem; }
+//	h3 { font-size: 1.3125rem; }
+//	h4 { font-size: 1.1875rem; }
+//	p { font-size: 1.0625rem; }
+//
```
Comment (kbeck303) [inline base/_typesizes.scss]: needs the closing media bracket }

File: base/_mixins.scss [hunk 3]
```diff
@@ -2,35 +2,106 @@
...
+@mixin touch-hover( $state: 'idle', $disabled: false ) {
+	$modernizr-hook-off: 'no-touchevents';
+	$modernizr-hook-on: 'touchevents';
+  $modernizr: true !default; // set this to false to maintain mixin but use the no-modernizr fallbacks
+
+	@if $state == 'idle' {
+		&,
+		&:link,
+		&:visited {
+		  @content;
```
Comment (kbeck303) [inline base/_mixins.scss]: the indents of `@content;` throughout this mixin are inconsistent

Comment (kbeck303) [reply base/_mixins.scss]: also tabs versus spaces in general throughout as well

**Comment by kbeck303** [comment]
I really like the "usage" and "return" examples for the mixins, that is super helpful (in my opinion).

**Comment by jhogue** [comment]
@kbeck303 I fixed that up, thanks for noticing.

**Comment by bendoh** [comment]
Still looks fine /approved

**Pull request merged**